lib/gmail-adapter.js

```diff
@@ -65,19 +84,41 @@ var Read = Juttle.proc.base.extend({
             if (messages && messages.length > 0) {
                 self.emit(messages);
             }
-            self.eof();
-        }).catch(function(err) {
+            return messages;
+        })
+        .then(function(messages) {
+            // If to is in the future, arrange with the scheduler to
+            // get all messages from the timestamp of the last message
+            // to "to". Otherwise we're done.
+            if (to === undefined ||
+                to.gt(now)) {
+                var next_poll = now.add(self.delay);
+                var next_from = from;
+
+                if (messages && messages.length > 0) {
+                    // XXX/mstemm is there a better way to express this?
```

Comment:
It might be more readable as 
```
var last_time = _.last(messages).time;
next_from = last_time.add(JuttleMoment.duration(1, 'ms')); 
```
but I don't think there's a fundamentally better way to express this.